vortex-array/src/validity.rs

```diff
@@ -211,6 +212,76 @@ impl Validity {
         Ok(validity)
     }
 
+    pub fn patch<P: AsPrimitive<usize>>(
+        self,
+        len: usize,
+        positions: &[P],
+        patches: Validity,
+    ) -> VortexResult<Self> {
+        if positions.last().map(|p| p.as_() >= len).unwrap_or(false) {
+            vortex_bail!("Position must be within length")
```

Comment:
nit, log values (preferably with the `OutOfBounds` error type that you can flag in the macro, too)